docs/site/tutorials/protocol_oriented_generics.ipynb

```diff
@@ -0,0 +1,1141 @@
+{
```

Comment:
<p>What is protocol-oriented programming? I think it would be useful to introduce this upfront to be able to highlight how the distinguishing features of protocol-oriented programming help writing quantifiably better software (more concise code, better performing code, etc).</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/swift/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>